imagetest/test_suites/oslogin/ssh_test.go

```diff
@@ -0,0 +1,322 @@
+//go:build cit
+// +build cit
+
+package oslogin
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// users
+	normalUser    = "normal-user"
+	adminUser     = "admin-user"
+	normal2FAUser = "normal-2fa-user"
+	admin2FAUser  = "admin-2fa-user"
+
+	// 2fa keys
+	normal2FAKey = "normal-2fa-key"
+	admin2FAKey  = "admin-2fa-key"
+
+	// keys
+	normalUserSshKey = "normal-user-ssh-key"
+	adminUserSshKey  = "admin-user-ssh-key"
+	normal2FASshKey  = "normal-2fa-ssh-key"
+	admin2FASshKey   = "admin-2fa-ssh-key"
+
+	// sudo keys
+	adminUserSudo = "admin-user-sudo"
+)
+
+// Changes the given metadata key to have the given value on the given instance.. If the key does not exist,
+// then this will create the key-value pair in the instance's metadata.
+func changeMetadata(ctx context.Context, client *compute.InstancesClient, key, value string) error {
+	vmname, err := getInstanceName(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting vm name: %v", err)
+	}
+
+	// Get project and zone of instance.
+	project, zone, err := getProjectZone(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Get instance info.
+	instancesGetReq := &computepb.GetInstanceRequest{
+		Instance: vmname,
+		Project:  project,
+		Zone:     zone,
+	}
+
+	instance, err := client.Get(ctx, instancesGetReq)
+	if err != nil {
+		return fmt.Errorf("error getting instance info: %v", err)
+	}
+	metadata := instance.Metadata
+
+	// Find the key in the metadata. If it doesn't exist, create a new metadata item.
+	found := false
+	for _, item := range metadata.Items {
+		if *(item.Key) == key {
+			item.Value = &value
+			found = true
+			break
+		}
+	}
+	if !found {
+		metadata.Items = append(metadata.Items, &computepb.Items{Key: &key, Value: &value})
+	}
+
+	// Update the metadata on the instance.
+	setMetadataReq := &computepb.SetMetadataInstanceRequest{
+		Instance:         vmname,
+		MetadataResource: metadata,
+		Project:          project,
+		Zone:             zone,
+	}
+	_, err = client.SetMetadata(ctx, setMetadataReq)
+	if err != nil {
+		return fmt.Errorf("error setting metadata: %v", err)
+	}
+	return nil
+}
+
+func sessionOSLoginEnabled(client *ssh.Client) error {
+	// We do not close the session as Run() implicitly closes the session after it's done running.
+	// Otherwise we run into an EOF error.
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create ssh session: %v", err)
+	}
+	data, err := session.Output("cat /etc/nsswitch.conf")
+	if err != nil {
+		return fmt.Errorf("failed to read /etc/nsswitch.conf: %v", err)
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
```

Comment:
This block is repeated, can we have a generic function to read these config files? The difference between this and the other I saw is the reader.